mindmeld/markup.py

```diff
@@ -270,13 +272,18 @@ def process_markup(markup, query_factory, query_options):
     return raw_text, query, entities
 
 
-def _process_annotations(query, annotations):
+def _process_annotations(query, annotations, sys_resolver=None):
     """
+    Args:
+        query (Query)
+        annotations (list)
+        sys_resolver (SystemEntityRecognizer): Default to Duckling
 
     Returns:
         list of ProcessedQuery:
     """
     stack = []
+    sys_resolver = sys_resolver or DucklingRecognizer.get_instance()
```

Comment:
I dont like this pattern of `sys_resolver = sys_resolver or DucklingRecognizer.get_instance()` everywhere since it doesnt isolate the responsibility of one class to instantiate the system_ent_recognizer. All other callers should be expecting a non-None system_ent_recognizer.

Reply:
the reason why I am doing this pattern is to make it the interface backward compatible; a non-backward compatible interface will necessitate a major release.

Reply:
fair point, but `_process_annotations ` is a private function by definition, so its not a public API.